packages/workspace/src/browser/workspace-frontend-contribution.ts

```diff
@@ -153,6 +153,14 @@ export class WorkspaceFrontendContribution implements CommandContribution, Keybi
             UriAwareCommandHandler.MonoSelect(this.selectionService, {
                 execute: (uri: URI) => this.saveAs(uri),
             }));
+        commands.registerCommand(WorkspaceCommands.OPEN_WORKSPACE_FILE, {
+            isEnabled: () => this.workspaceService.saved,
+            execute: () => {
+                if (this.workspaceService.saved && this.workspaceService.workspace) {
```

Comment:
Looks like the check for ` this.workspaceService.workspace` is redutdant: `this.workspaceService.saved` contains this check https://github.com/eclipse-theia/theia/blob/85aa2d5dba8f9b674d896c300f7d3a36016a7e00/packages/workspace/src/browser/workspace-service.ts#L524

Reply:
Fair point. I think I had them in the other order originally, but that wouldn't make it less redundant 😄

Reply:
Actually it turns out it was a typeguard. Without that check, Typescript isn't sure that `this.workspacesService.workspace.resource` (the URI for the workspace file) exists, so I'm going to leave it in.